tests/test_utils.py

```diff
@@ -306,4 +310,46 @@ def test_init_out_dir(tmp_path):
 
     with mock.patch.dict(os.environ, {"LIGHTNING_ARTIFACTS_DIR": "prefix"}):
         assert init_out_dir(relative_path) == Path("prefix") / relative_path
-        assert init_out_dir(absolute_path) == absolute_path
\ No newline at end of file
+        assert init_out_dir(absolute_path) == absolute_path
+
+
+@pytest.fixture
+def model_parameters():
+    return [torch.nn.Parameter(torch.randn(2, 2))]
+
+
+def test_instantiate_bnb_optimizer_with_str(model_parameters):
+    import bitsandbytes as bnb
+    with mock.patch("litgpt.utils.get_argument_names", return_value={"lr", "eps", "weight_decay"}):
+        optimizer = instantiate_bnb_optimizer("AdamW", model_parameters)
+        assert isinstance(optimizer, bnb.optim.adamw.PagedAdamW)
+
+
+def test_instantiate_bnb_optimizer_with_dict(model_parameters):
+    import bitsandbytes as bnb
+    optimizer_dict = {"class_path": "AdamW", "init_args": {"lr": 0.01}}
+    with mock.patch("litgpt.utils.get_argument_names", return_value={"lr", "eps", "weight_decay"}):
+        optimizer = instantiate_bnb_optimizer(optimizer_dict, model_parameters)
+        assert isinstance(optimizer, bnb.optim.adamw.PagedAdamW)
+        assert optimizer.param_groups[0]["lr"] == 0.01
+
+
+def test_instantiate_bnb_optimizer_with_invalid_str(model_parameters):
+    with pytest.raises(ValueError):
```

Comment:
Always add the "match" message in case you eat up an undesired error

```suggestion
    with pytest.raises(ValueError, match="only supports the AdamW"):
```